flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java

```diff
@@ -199,16 +199,23 @@ private String toClassName(String resourceName) {
                     }
 
                     try {
-                        boolean directoryFound = false;
-                        Enumeration<JarEntry> entries = jarFile.entries();
-                        while (entries.hasMoreElements()) {
-                            if (entries.nextElement().isDirectory()) {
-                                directoryFound = true;
-                                break;
+                        // Look up resources from jar files within an Ant task
+                        boolean shouldLookupResources = classLoader.getParent() != null
+                                            && "AntClassLoader".equals(classLoader.getParent().getClass().getSimpleName());
```

Comment:
I am not sure this is the right place to detect the Ant ClassLoader. Why is this even necessary anyway? This fallback code is really just for broken jar files missing directory entries.